### PR TITLE
MED-102 Implement Booking 1 UI

### PR DESCRIPTION
**Pull request by SoulOfDarknes**
1. Add IU 1 step Booking [Appointments]
2. Add patient info
**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

first page
image

when you pick day
image

choose time
image

select doctor
image

### CODE REVIEW

File: apps/medon-fe/src/components/Steps/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  border-radius: 4px;
+  background: ${theme.colors.blue_300};
+  padding: 8px 14px 8px 25px;
+  font-family: ${theme.fontFamily.roboto};
+  font-style: normal;
+  font-weight: ${theme.fontWeight.bold};
+  font-size: ${theme.fontSizes.md};
+  line-height: 20px;
+  color: #ffffff;
+  background-image: url(/assets/svgs/cross.svg);
```
Comment (skazzp) [inline apps/medon-fe/src/components/Steps/styles.tsx]: path to const

File: apps/medon-fe/src/components/Steps/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        ? '90px'
+        : buttonType === 'previous'
+        ? '5px'
+        : '155px'};
+  }
+
+  ${(props) =>
+    props.buttonType === 'next' &&
+    `
+    background-image: url(/assets/svgs/appointments/arrowRight.svg);
```
Comment (skazzp) [inline apps/medon-fe/src/components/Steps/styles.tsx]: same

**Review by skazzp — CHANGES_REQUESTED** [review]

File: apps/medon-fe/src/components/Steps/styles.tsx (hunk 1 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Steps/styles.tsx]: done

File: apps/medon-fe/src/components/Steps/styles.tsx (hunk 2 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Steps/styles.tsx]: done

File: apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  const { formats } = useMemo(
+    () => ({
+      formats: {
+        weekdayFormat: (
+          date: Date,
+          culture: string | undefined,
+          localizer: DateLocalizer | undefined
+        ) => {
+          if (localizer) {
+            return localizer.format(date, 'dddd', culture);
```
Comment (skazzp) [inline apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx]: date format to const

**Review by skazzp — CHANGES_REQUESTED** [review]

File: apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx (hunk 3 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx]: done

**Review by skazzp — APPROVED** [review]

File: apps/medon-fe/src/components/Steps/index.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      case steps.two:
+        return <div>{t('patient-info.time')}</div>;
+      case steps.three:
+        return <div>{t('patient-info.remote-doctor')}</div>;
+      default:
+        return null;
+    }
+  };
+
+  const getDoctorFullName = (doctorId: number) => {
```
Comment (lyudad) [inline apps/medon-fe/src/components/Steps/index.tsx]: it will be called with each rerender, does it ok?

Comment (lyudad) [reply apps/medon-fe/src/components/Steps/index.tsx]: whare is output types?

File: apps/medon-fe/src/components/Steps/styles.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  font-family: ${theme.fontFamily.roboto};
+  font-style: normal;
+  font-weight: ${theme.fontWeight.bold};
+  font-size: ${theme.fontSizes.md};
+  line-height: 20px;
+  color: ${theme.colors.white};
+  &:hover {
+    background-color: ${theme.colors.blue_700};
+    background-position-x: ${({ buttonType }) =>
+      buttonType === 'next'
```
Comment (lyudad) [inline apps/medon-fe/src/components/Steps/styles.tsx]: we are con compare with text

File: apps/medon-fe/src/components/Steps/index.tsx (hunk 4 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Steps/index.tsx]: fixed rewrites and put it to different file

File: apps/medon-fe/src/components/Steps/styles.tsx (hunk 5 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Steps/styles.tsx]: fixed

File: apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  setSelectedDate,
+  selectedDate,
+}: BookAppointmentCalendarProps) {
+  const { formats } = useMemo(
+    () => ({
+      formats: {
+        dateFormat: dateFormatCalendar,
+        weekdayFormat: (
+          date: Date,
+          culture: string | undefined,
```
Comment (lyudad) [inline apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx]: culture?: string should work

Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/BookAppointmentCalendar/index.tsx]: fixed

**Pull request merged**